pkg/controller/installation/products/rhsso/reconciler.go

```diff
@@ -163,9 +176,37 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, inst *v1alpha1.Ins
 		},
 	}
 	ownerutil.EnsureOwner(kcr, inst)
-	err = serverClient.Create(ctx, kcr)
-	if err != nil && !k8serr.IsAlreadyExists(err) {
-		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to create keycloak realm")
+	_, err = controllerutil.CreateOrUpdate(ctx, serverClient, kcr, UpdateNoOp)
```

Comment:
Why switch to `CreateOrUpdate` if we are doing nothing on update?

Reply:
I want the update as I want to update:
```go
CreateOnly:                        false,
DeleteUsers:                       true,
```
and in the case of an already existing CR (as in an upgrade scenario) the CR wouldn't get updated.

I believe that `CreateOrUpdate` does an equality check and updates based on the passed obj anyway but let me check